src/KickMCMC.jl

```diff
@@ -11,6 +20,83 @@ end
 Distributions.logpdf(d::WrappedCauchy, x::Real) = log(1/(2*π)*(sinh(d.σ)))-log(cosh(d.σ)-cos(x-d.μ))
 Distributions.pdf(d::WrappedCauchy, x::Real) = 1/(2*π)*(sinh(d.σ))/(cosh(d.σ)-cos(x-d.μ))
 
+"""
+    mutable struct Observations
+
+Observations contains the symbols, values, errors, and units of each observed parameter.
+"""
+@kwdef mutable struct Observations
+    props::Vector{Symbol}
+    vals::Vector{Float64}
+    errs::Vector{Float64}
+    units::Vector{Float64}
+end
+
+"""
+    mutable struct Priors
+
+Priors contains the prior distribution of each of the desired parameters
+"""
+@kwdef mutable struct Priors
+    logm1_dist::ContinuousUnivariateDistribution
+    logm2_dist::ContinuousUnivariateDistribution
+    logP_dist::ContinuousUnivariateDistribution 
+    vkick_dist::ContinuousUnivariateDistribution
+    frac_dist::ContinuousUnivariateDistribution 
+end
+
+"""
+    mutable struct KickMCMCResults
+
+KickMCMCResults contains the MCMC model and Observations structs, the Results dict,
+the chains that resulted from the MCMC, and the parameters that went into the sampler.
+"""
+@kwdef mutable struct KickMCMCResults
+    # TODO RTW: do I need types for everything? what do I use if non-trivial?
+    mcmc_model # RTW todo
+    observations::Observations
+    results::Dict{Symbol, Matrix{Float64}}
+    chains # RTW todo
+    nuts_warmup_count::Int
+    nuts_acceptance_rate::Float64
+    nsamples::Int
+end
+
+
+
+
+
+########################
+### 
+### Function definitions
+### 
+########################
+
+# RTW TODO
+
+function createObservations(obs_matrix::Vector{Vector{Any}})
+    obs_matrix = stack(obs_matrix) # make into actual matrix
+    return Observations(
+        props = obs_matrix[1,:],
+        vals  = obs_matrix[2,:],
+        errs  = obs_matrix[3,:],
+        units = obs_matrix[4,:])
+end
+
+function createPriors(;
+    logm1_dist = Uniform(0.1,3), # in log(Msun)
+    logm2_dist = Uniform(0.1,3), # in log(Msun)
+    logP_dist  = Uniform(-1,3),   # in log(days)
+    vkick_dist = Exponential(1), # in 100 km/s
+    frac_dist  = Uniform(0,1.0))
+    return Priors(
+        logm1_dist = logm1_dist,  
+        logm2_dist = logm2_dist,  
+        logP_dist  = logP_dist ,  
+        vkick_dist = vkick_dist,  
+        frac_dist  = frac_dist )
+end
```

Comment:
Both of these constructors seem like unnecessary wrapping of the one that is created by default

Reply:
The purpose is to make the arguments optional. It's not clear to me how to do that without the wrappers.